homeassistant/components/nibe_heatpump/climate.py

```diff
@@ -0,0 +1,220 @@
+    if not main_unit:
+        LOGGER.debug("Skipping climates - no main unit found")
+        return
+
+    def climate_systems():
+        for key, group in CLIMATE_COILGROUPS.get(coordinator.series, ()).items():
+            try:
+                yield NibeClimateEntity(coordinator, key, main_unit, group)
+            except CoilNotFoundException as exception:
+                LOGGER.debug("Skipping climate: %s due to %s", key, exception)
+
+    async_add_entities(climate_systems())
+
+
+class NibeClimateEntity(CoordinatorEntity[Coordinator], ClimateEntity):
+    """Climate entity."""
+
+    _attr_entity_category = None
+    _attr_supported_features = (
+        ClimateEntityFeature.TARGET_TEMPERATURE_RANGE
+        | ClimateEntityFeature.TARGET_TEMPERATURE
+    )
+    _attr_hvac_modes = [HVACMode.HEAT_COOL, HVACMode.OFF, HVACMode.HEAT]
+    _attr_target_temperature_step = 0.5
+    _attr_max_temp = 35.0
+    _attr_min_temp = 5.0
+
+    def __init__(
+        self,
+        coordinator: Coordinator,
+        key: str,
+        unit: UnitCoilGroup,
+        climate: ClimateCoilGroup,
+    ) -> None:
+        """Initialize entity."""
+        super().__init__(
+            coordinator,
+            {
+                unit.prio,
+                unit.cooling_with_room_sensor,
+                climate.current,
+                climate.setpoint_heat,
+                climate.setpoint_cool,
+                climate.mixing_valve_state,
+                climate.active_accessory,
+                climate.use_room_sensor,
+            },
+        )
+        self._attr_available = False
+        self._attr_name = climate.name
+        self._attr_unique_id = f"{coordinator.unique_id}-{key}"
+        self._attr_device_info = coordinator.device_info
+        self._attr_hvac_action = HVACAction.IDLE
+        self._attr_hvac_mode = HVACMode.OFF
+        self._attr_target_temperature_high = None
+        self._attr_target_temperature_low = None
+        self._attr_target_temperature = None
+        self._attr_entity_registry_enabled_default = climate.active_accessory is None
+
+        def _get(address: int) -> Coil:
+            return coordinator.heatpump.get_coil_by_address(address)
+
+        self._coil_current = _get(climate.current)
+        self._coil_setpoint_heat = _get(climate.setpoint_heat)
+        self._coil_setpoint_cool = _get(climate.setpoint_cool)
+        self._coil_prio = _get(unit.prio)
+        self._coil_mixing_valve_state = _get(climate.mixing_valve_state)
+        if climate.active_accessory is None:
+            self._coil_active_accessory = None
+        else:
+            self._coil_active_accessory = _get(climate.active_accessory)
+        self._coil_use_room_sensor = _get(climate.use_room_sensor)
+        self._coil_cooling_with_room_sensor = _get(unit.cooling_with_room_sensor)
+
+        if self._coil_current:
+            self._attr_temperature_unit = self._coil_current.unit
+
+    def _handle_coordinator_update(self) -> None:
```

Comment:
Please decorate this method with `@callback` as in the parent coordinator entity.

Reply:
https://github.com/home-assistant/core/pull/83027